src/wallet/mod.rs

```diff
@@ -485,13 +493,28 @@ where
         // the new tx must "pay for its bandwidth"
         let vbytes = tx.get_weight() as f32 / 4.0;
         let required_feerate = FeeRate::from_sat_per_vb(details.fees as f32 / vbytes + 1.0);
-        let new_feerate = builder.fee_rate.unwrap_or_default();
-
-        if new_feerate < required_feerate {
-            return Err(Error::FeeRateTooLow {
-                required: required_feerate,
-            });
-        }
+        let new_feerate = match builder
+            .fee_policy
+            .as_ref()
+            .unwrap_or(&FeePolicy::FeeRate(FeeRate::default()))
+        {
+            FeePolicy::FeeAmount(amount) => {
+                if *amount < details.fees {
+                    return Err(Error::FeeTooLow {
+                        required: details.fees,
+                    });
+                }
+                FeeRate::from_sat_per_vb(0.0)
+            }
+            FeePolicy::FeeRate(rate) => {
+                if *rate < required_feerate {
+                    return Err(Error::FeeRateTooLow {
+                        required: required_feerate,
+                    });
+                }
+                *rate
+            }
+        };
```

Comment:
I think this `match` can be merged with the one below that initializes `initial_fee`. I would do it mostly for consistency with `create_tx` and to avoid duplicating the `.unwrap_or(...)` logic